### ISSUE TITLE
Write/Revert Changes remain enabled if SELECT statement is terminated with a semicolon and whitespace follows

### ISSUE THREAD

**Issue by MikeDimmickMnetics** [sqlitebrowser/sqlitebrowser#2147]
If the SELECT query is terminated with a semicolon, but is followed by whitespace, or a comment, the transaction that wraps the query is not committed or rolled back. This means that the Write Changes and Revert Changes buttons remain enabled, and the database is locked for changes from other processes.

#### What did you do?

This can be reproduced simply by running:

```
SELECT * FROM sqlite_master;

```

(include the newline after the semicolon).

It can also be reproduced with the following:

```
SELECT * FROM sqlite_master;/**/
```

#### What did you expect to see?

The Write Changes and Revert Changes buttons should be disabled after running the query.

#### What did you see instead?

The Write Changes and Revert Changes buttons remain enabled. It is necessary to click one of the buttons to release the database for other processes.

This problem does not occur if the statement is not terminated with a semicolon. The problem does not occur if query text is selected, and the selection ends with the semicolon. Leading whitespace, before the SELECT, does not cause the problem.

Useful extra information
-------------------------
> DB4S v3.11.2 [built for x86_64-little_endian-llp64] on Windows 10 (10.0) (winnt/10.0.18362) [x86_64]
> using SQLite Version 3.27.2
> and Qt 5.11.3

**Comment by justinclift** [comment]
Interesting bug. :smile:

As a thought, since we're getting close to releasing a new version... any interest in seeing if the bug still shows up in our nightly builds?

&nbsp; &nbsp; https://nightlies.sqlitebrowser.org/latest/

**Comment by mgrojo** [comment]
Yes, I think it's fixed in the nightly. @MikeDimmickMnetics Could you confirm, so we can mark this issue as fixed in our next release?

**Comment by MikeDimmickMnetics** [comment]
Yes, this is fixed in the latest nightly (3.11.99 Mar 3 2020, x86_64 on Windows). Thanks for your help.

**Comment by justinclift** [comment]
Awesome.  Thanks for checking it @MikeDimmickMnetics and letting us know. :smile:
